source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -339,9 +339,8 @@ bool InstanceImplPosix::illegalPath(const std::string& path) {
   // platform in the future, growing these or relaxing some constraints (e.g.
   // there are valid reasons to go via /proc for file paths).
   // TODO(htuch): Optimize this as a hash lookup if we grow any further.
-  if (absl::StartsWith(canonical_path.return_value_, "/dev") ||
-      absl::StartsWith(canonical_path.return_value_, "/sys") ||
-      absl::StartsWith(canonical_path.return_value_, "/proc")) {
+  if (canonical_path.return_value_ == "/dev" || canonical_path.return_value_ == "/sys" ||
```

Comment:
I think this now allows the file to be at "/dev/abc/my_file" which is what this check intended to avoid in the first place.
The check could be for startwith("/sys/") or equals("/sys"). I believe this should align the intentions of #3165 and your need to support the "/sysroot" directory.
cc @htuch as the author of #3165.

Reply:
Thanks @adisuissa , I will update it.